### ISSUE TITLE
is it possbile to compile GOTURN on windows?

### ISSUE THREAD

**Issue by BackT0TheFuture** [davheld/GOTURN#28]

**Comment by joandrade** [comment]
I've been trying to compile this in Windows for the past three weeks, and the answer is: yes. It will compile eventually under Visual Studio. The problem I'm having now is at runtime. I get:

> F0303 17:51:30.612993  4496 layer_factory.cpp:62] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Input (known types: Convolution, Eltwise, LRN, Pooling, Power, ReLU, Sigmoid, Softmax, Split, TanH)

This error which suggests that Caffe is not playing nice with GOTURN under Windows. I've read that one of the solutions ([seen here](http://blog.csdn.net/fangjin_kl/article/details/50936952)) is to actually add the GOTURN project to the Caffe solution, and all its dependencies, and the layer registration will work properly in those conditions. I'm not familiar with neural networks (I was just tasked with porting this algorithm) and this solution is impractical for my implementation, so that's where I stopped. Maybe @davheld can chime in about this error, since I seem to be the one person who made it this far.

Cheers.

**Comment by davheld** [comment]
"Input" is an unknown layer type?  How do you normally provide inputs to your network in Windows?  I might try to play around more with getting networks to run in Windows with Caffe (not using GOTURN) and to see how the inputs are normally provided, and then use that to figure out what is wrong.  Sorry I don't have more advice on this.

**Comment by joandrade** [comment]
I don't know. Like I said, I was tasked to see how far I could take this to be implemented in a Windows platform. I'm currently looking into the solution I linked in my first post, and see how feasible it is to implement with our current software. That's okay, thank you for your input.
